pkg/virt-handler/spice-proxy/spice-proxy.go

```diff
@@ -0,0 +1,224 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 Red Hat, Inc.
+ *
+ */
+package spice_proxy
+
+//go:generate mockgen -source $GOFILE -package=$GOPACKAGE -destination=generated_mock_$GOFILE
+
+/*
+ ATTENTION: Rerun code generators when interface signatures are modified.
+*/
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"sync"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/log"
+	"kubevirt.io/kubevirt/pkg/virt-handler/isolation"
+)
+
+var (
+	proxyRange = []int{30000, 32000}
+)
+
+type UnixToTcpProxyManager interface {
+	StartListener(vmNamespace, vmName string, vmi *v1.VirtualMachineInstance) error
+	GetPort(vmNamespace, vmName string) (int32, error)
+	StopListener(vmNamespace, vmName string)
+}
+
+type spiceProxyManager struct {
+	proxies              map[string]*spiceProxy
+	allocatedPort        map[int32]bool
+	managerLock          sync.Mutex
+	podIsolationDetector isolation.PodIsolationDetector
+}
+
+type spiceProxy struct {
+	unixSocketPath string
+	tcpBindAddress string
+	tcpBindPort    int32
+	stopChan       chan struct{}
+	listenErrChan  chan error
+	fdChan         chan net.Conn
+	listener       net.Listener
+}
+
+func NewSpiceProxyManager(podIsolationDetector isolation.PodIsolationDetector) UnixToTcpProxyManager {
+	return &spiceProxyManager{proxies: make(map[string]*spiceProxy), allocatedPort: make(map[int32]bool), podIsolationDetector: podIsolationDetector}
+}
+
+func (spm *spiceProxyManager) StartListener(vmNamespace, vmName string, vmi *v1.VirtualMachineInstance) error {
+	spm.managerLock.Lock()
+	defer spm.managerLock.Unlock()
+
+	if _, isExist := spm.proxies[uniqueVmName(vmNamespace, vmName)]; isExist {
```

Comment:
can `isExist` be just `exists`? Or even `ok`? The scope is so narrow that the name does not have to be long

Reply:
Normally `exists` for checking if a key exists, `ok` if a type conversion works?

Reply:
I happened to see ok also for checking if a key exists, see for example https://blog.golang.org/go-maps-in-action .

Reply:
Sure, I would never ask to change it. More a tendency.

Reply:
Agree, would have preferred `exists` over `isExist` though

Reply:
done